File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode swapPairs(ListNode head) {
        if(head==null || head.next==null)
        return head;
         ListNode d1=new ListNode (-1);
    ListNode t=d1;
    ListNode i=head;
    while(i!=null && i.next!=null)
    {
        ListNode j=i.next;
        ListNode k=j.next;

        t.next=j;
        j.next=i;
        i.next=k;

        t=i;
        i=k;
    }
    return d1.next;
    }
}

File: 0024-swap-nodes-in-pairs/0024-swap-nodes-in-pairs.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
    */
    struct ListNode* swapPairs(struct ListNode* head) {
        if(head==NULL || head->next==NULL)
        return head;
    struct ListNode* d1=malloc(sizeof(struct ListNode));
        d1->val=-1;
        d1->next=NULL;
    struct ListNode* t=d1;
    struct ListNode* i=head;
    while(i!=NULL && i->next!=NULL)
    {

    struct ListNode* next = i->next->next;
        t->next=i->next;
        t=t->next;
        t->next=i;
        t=t->next;
        i=next;
    }
    t->next=i;
    return d1->next;
    }